### ISSUE TITLE
Test Ramp with manifests with very large structures

### ISSUE THREAD

**Issue by elynema** [samvera-labs/ramp#196]
# Description

Media Collections Online (Avalon instance @ IU) contains numerous media items with very large structures (600+ canvases, for example). IIIF manifests like this will be very long with tons of nesting. Test Ramp against some of these records to determine if it is able to parse and display the manifest in a reasonable amount of time, or if we need to optimize the manifest parsing. Ramp will also need to be able to handle these before it can integrate with Avalon and the many-sectioned items that are standard for IU Avalon instances.

# Questions
- Will work done to speed up media object page load with large structures for Avalon also apply when creating manifests for the same items with large structures?
- Will this work well with switching streams between canvases? HLS streaming takes a bit to load the media player; may need to block UI actions until media loads.
- What things can we look to to increase parsing speed?
- What IIIF Commons libraries are available to help with parsing?
- We should be able to discard time information about fetching the manifest rather than parsing (fetching is arbitrary and will vary; should also be testing in general for Avalon)
- Ramp should be able to handle any amount of wait for fetching a manifest (large dynamic manifests which will take time to generate)
- Ramp may retrieve the manifest at a later point as well

# Examples
- https://media.dlib.indiana.edu/media_objects/gq67js70q (24 sections)
- https://media.dlib.indiana.edu/media_objects/79407z521 (271 sections)

# Done Looks Like
- [ ] Document current speed for Ramp to parse manifests with many sections
- [ ] Identify slow areas and create new tickets as needed

**Comment by joncameron** [comment]
avalon-staging may have objects that can be used for this testing.

**Comment by joncameron** [comment]
avalon-dev could use more items for broader testing (such as items with many sections and transcripts).

**Comment by joncameron** [comment]
@joncameron to work on populating avalon-dev with test records.

**Comment by joncameron** [comment]
Refactor work in https://github.com/samvera-labs/ramp/issues/226

**Comment by cjcolvar** [comment]
FWIW https://github.com/avalonmediasystem/avalon/issues/5476 sped up manifest generation on avalon-dev for a 200 section item from 40s to 8s.

**Comment by joncameron** [comment]
I'm going to close this one out now. Final thoughts:
- Ramp seems to parse things quickly enough even for large manifests. Parsing time is low enough to not be a concern.
- Manifest generation time and Item page load time are two much larger areas where focus should be for improvement. As Chris noted, manifest generation time got significantly better but it's still a long wait on the page for users. I'll create a new issue to look at what continued improvements could be made for generation time.
